### PR TITLE
Return the correct zone if the use uses abrev zones

### PR DESCRIPTION
**Pull request by garthy**
The user can use abbreviated zones when creating volumes such as "AD-1" to make the yaml portable across zone (failure-domain.beta.kubernetes.io/zone) . This fixes an issue where the abbreviated zone was used on the volume label and caused any pod that tried to use it as unschedulable.

### CODE REVIEW

File: pkg/provisioner/core/provisioner.go [hunk 1]
```diff
@@ -91,6 +91,16 @@ func roundUpSize(volumeSizeBytes int64, allocationUnitBytes int64) int64 {
 	return (volumeSizeBytes + allocationUnitBytes - 1) / allocationUnitBytes
 }
 
+// mapAvailabilityDomainToFailureDomain maps a given Availability Domain to a
+// k8s label compat. string.
+func mapAvailabilityDomainToFailureDomain(AD string) string {
```
Comment (owainlewis) [inline pkg/provisioner/core/provisioner.go]: Could do with a unit test.

**Comment by owainlewis** [comment]
Change looks good. Suggest adding a unit test to clarify what mapAvailabilityDomainToFailureDomain is supposed to do depending on different inputs and catch any potential edge cases that might not have been considered.
